elastic/kibana
# [Security Solution] Show correct `Privileges Required` page for unauthorized paths

## Summary

The goal is to show a _Privileges required_  or _Superuser required_ message to users if they want to access a page in Security Management they don't have access for, based on feature flags.

`<PrivilegedRouter>` decides which message to show:
- current version (no feature flags): _superuser required_ for all pages
- v1 (`endpointRbacV1Enabled`):
   - _privileges required_ only for _Response actions history_
   - _superuser required_ for other pages
- v2 (`endpointRbacEnabled`): _privileges required_ for all pages

To test it:
- change the feature flags,
- create a non-superuser user, with different Kibana privileges
- and check the urls for Security Management pages, e.g. http://localhost:5601/app/security/administration/response_actions_history

Without feature flags:
<img width="998" alt="image" src="https://user-images.githubusercontent.com/39014407/200780212-e16246a0-76d4-4c8a-a9b9-6b2b4466e7b8.png">

v1/v2 for Response actions history:
<img width="1006" alt="image" src="https://user-images.githubusercontent.com/39014407/200780408-c80ca58b-9d06-47cd-beb9-c764ef5f1380.png">


### Dev note
A refactor PR for reusing the new `<NoPrivilegesPage>` component in 3 other places inside security solution plugin is ~coming soon.~ here: https://github.com/elastic/kibana/pull/144886

### Checklist

Delete any items that are not applicable to this PR.
